vimiv/imutils/exif.py

```diff
@@ -10,139 +10,361 @@
-        super().__init__()
-        self._exif = None
+    if pyexiv2:
+        return handler
+
+    if piexif:
+        return _ExifHandlerPiexif
+
+    _logger.warning(
+        "There is no exif support and therfore: \
+1. Exif data is lost when writing images to disk; \
+2. The `:metadata` command and associated `i` keybinding is not available; \
+3. The {exif-date-time} statusbar module is not available"
+    )
 
-        if piexif is None:
-            log.error("%s relies on exif support", self.__class__.__qualname__)
-        else:
-            self._load_exif(filename)
+    return _ExifHandlerNoExif
+
+
+@check_exif_dependancy
+class ExifHandler(_ExifHandler):
+    """Main ExifHandler implementation bases on py3exiv2."""
+
+    def __init__(self, filename=""):
 
-    def _load_exif(self, filename):
-        """Load exif information from filename into the dictionary."""
         try:
-            self._exif = piexif.load(filename)
-            desired_keys = [
-                e.lower().strip()
-                for e in api.settings.metadata.current_keyset.value.split(",")
-            ]
-            _logger.debug(f"Read metadata.current_keys {desired_keys}")
-        except (piexif.InvalidImageDataError, FileNotFoundError, KeyError):
+            self._metadata = pyexiv2.ImageMetadata(filename)
+            self._metadata.read()
+
+        except FileNotFoundError:
+            _logger.debug("File %s not found", filename)
             return
 
-        for ifd in self._exif:
-            if ifd == "thumbnail":
-                continue
-
-            for tag in self._exif[ifd]:
-                keyname = piexif.TAGS[ifd][tag]["name"]
-                keytype = piexif.TAGS[ifd][tag]["type"]
-                val = self._exif[ifd][tag]
-                _logger.debug(
-                    f"name: {keyname} type: {keytype} value: {val} tag: {tag}"
-                )
-                if keyname.lower() not in desired_keys:
-                    _logger.debug(f"Ignoring key {keyname}")
+    def get_formatted_exif(self) -> Dict[str, str]:
+
+        desired_keys = [
+            e.strip() for e in api.settings.metadata.current_keyset.value.split(",")
+        ]
+        _logger.debug(f"Read metadata.current_keys {desired_keys}")
+
+        exif = dict()
+
+        for base_key in desired_keys:
+            # For backwards compability, assume it has one of the following prefixes
+            for prefix in ["", "Exif.Image.", "Exif.Photo."]:
+                key = f"{prefix}{base_key}"
+                try:
+                    key_name = self._metadata[key].name
+
+                    try:
+                        key_value = self._metadata[key].human_value
+
+                    # Not all metadata(iptc) provide human_value, take raw_value instead
+                    except AttributeError:
+                        key_value = self._metadata[key].raw_value
+
+                    exif[key] = (key_name, key_value)
```

Comment:
Here is where you first introduce the new maping type, this comment is just for reference and possibly adding the explanation here if it has to do with the py3exiv2 implementation.